### ISSUE TITLE
Generated ssh key "does not appear to be a valid SSH private key".

### ISSUE THREAD

**Issue by ShadowMitia** [ionic-team/ionic-cli#3688]
**Description:**

I'm trying to link my project to ionic pro, and I've done the steps to add an ssh key with the ionic CLI.
All the steps work fine except the last one that fails on this error: "[FILE] does not appear to be a valid SSH private key. (Missing '-----BEGIN RSA PRIVATE KEY-----' header.)".
[FILE] being a valid public key file, but the header is not the same: "-----BEGIN OPENSSH PRIVATE KEY-----"

**Steps to Reproduce:**

The steps I did:
- ionic link
and following the steps

Also tried with
- ionic setup ssh

After having the ssh keys generated it will always fail on:
- ionic setup ssh use "path_to_pub_file"

**Output:**

 `ionic ssh use ../../../.ssh/ionic/FILE.pub`
```
[ERROR] ../../../.ssh/ionic/FILE.pub does not appear to be a valid SSH private key. (Missing '-----BEGIN RSA PRIVATE 
        KEY-----' header.)
        
        If you are having issues, try using ionic ssh setup.

```

**My `ionic info`:**

```
✔ Gathering environment info - done!

Ionic:

   ionic (Ionic CLI)  : 4.2.1 (/usr/lib/node_modules/ionic)
   Ionic Framework    : ionic-angular 3.9.2
   @ionic/app-scripts : 3.2.0

Cordova:

   cordova (Cordova CLI) : 8.1.2 (cordova-lib@8.1.1)
   Cordova Platforms     : android 7.0.0, browser 5.0.4
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.2, cordova-plugin-ionic-webview 2.1.4, (and 7 other plugins)

System:

   Android SDK Tools : 26.1.1 (/home/dimitri/Android/Sdk)
   NodeJS            : v8.12.0 (/usr/bin/node)
   npm               : 6.4.1
   OS                : Linux 4.18

```

**Other Information:**

ssh version: `OpenSSH_7.8p1, OpenSSL 1.1.0i-fips  14 Aug 2018`

**Comment by dwieeb** [comment]
@ShadowMitia Looks like the format of these keys changed in OpenSSH 7.8. I know it works with 7.6. The workaround is to downgrade.

**Comment by ShadowMitia** [comment]
Downgrade OpenSSH or Ionic? :grin: 

So the problem is really just a one word change in the headers?

**Comment by dwieeb** [comment]
Yeah. We do some validation so that people choose the correct file. Otherwise they'll get weird, obscure errors later on.

Downgrade OpenSSH to 7.6 is the workaround.

**Comment by dwieeb** [comment]
@ShadowMitia This issue should be fixed in CLI 4.3.0!
